src/tests/frontend/helper.js

```diff
@@ -28,20 +15,26 @@ const helper = {};
     return randomstring;
   };
 
-  const getFrameJQuery = ($iframe) => {
+  const getFrameJQuery = async ($iframe, includeJquery = false, includeSendkeys = false) => {
     /*
       I tried over 9001 ways to inject javascript into iframes.
       This is the only way I found that worked in IE 7+8+9, FF and Chrome
     */
     const win = $iframe[0].contentWindow;
     const doc = win.document;
 
-    // IE 8+9 Hack to make eval appear
-    // https://stackoverflow.com/q/2720444
-    win.execScript && win.execScript('null');
-
-    win.eval(jsLibraries.jquery);
-    win.eval(jsLibraries.sendkeys);
+    if (!win.$ && includeJquery) {
+      const scriptElem = doc.createElement('script');
+      scriptElem.setAttribute('src', '../../static/js/vendors/jquery.js');
+      doc.head.appendChild(scriptElem);
+      await helper.waitForPromise(() => win.$, 5000);
+    }
+    if (!win.bililiteRange && includeSendkeys) {
+      const scriptElem = doc.createElement('script');
+      scriptElem.setAttribute('src', '../tests/frontend/lib/sendkeys.js');
+      doc.head.appendChild(scriptElem);
+      await helper.waitForPromise(() => win.$ && win.bililiteRange, 5000);
+    }
```

Comment:
My preference would be to create a Promise that resolves when the `load` event fires on the script element or rejects when the `error` event fires, though that is more complicated.

For example (untested):
```suggestion
    const load = async (url) => {
      const elt = doc.createElement('script');
      elt.setAttribute('src', url);
      const p = new Promise((resolve, reject) => {
        const handler = (ev) => {
          elt.removeEventListener('load', handler);
          elt.removeEventListener('error', handler);
          if (ev.type !== 'load') return reject(new Error(`failed to load ${url}`));
          resolve();
        };
        elt.addEventListener('load', handler);
        elt.addEventListener('error', handler);
      });
      doc.head.appendChild(elt);
      await p;
    };
    if (!win.$ && includeJquery) await load('../../static/js/vendors/jquery.js');
    // sendkeys.js depends on jQuery, so it cannot be loaded until jQuery has finished loading.
    if (!win.bililiteRange && includeSendkeys) await load('../tests/frontend/lib/sendkeys.js');
```

Reply:
Will do this. Not sure if there is some delay when loading one of the two files, so that there is any chance `win.$` or `win.bililiteRange` is defined, but not fully loaded yet, but your suggestion will prevent this from happening, so it's the better solution.

Reply:
done